File: articles/trauma-dissociation-and-the-fragmented-psyche/fortran/trauma_integration.f90
! Trauma, Dissociation, and the Fragmented Psyche
! Fortran example: simple synthetic integration potential score

program trauma_integration_demo
  implicit none

  real :: score

  score = 0.60 * 0.82 + 0.55 * 0.80 + 0.62 * 0.84 + &
          0.48 * 0.80 + 0.44 * 0.78 + 0.40 * 0.76 + &
          0.32 * 0.78 - 0.68 * 0.24 - 0.34 * 0.28

  print *, "Synthetic trauma integration potential:", score
end program trauma_integration_demo

File: articles/trauma-dissociation-and-the-fragmented-psyche/c/trauma_integration.c
/*
Trauma, Dissociation, and the Fragmented Psyche
C example: simple synthetic integration potential score

Synthetic educational demonstration only.
Not for diagnosis, therapy, crisis assessment, risk prediction,
screening, surveillance, or individual evaluation.
*/

#include <stdio.h>

double integration_potential(
    double ego_integration,
    double symbolic_capacity,
    double relational_safety,
    double bodily_regulation,
    double memory_continuity,
    double symbolic_recovery,
    double witnessing_capacity,
    double dissociation,
    double nightmare_intrusion
) {
    return 0.60 * ego_integration
         + 0.55 * symbolic_capacity
         + 0.62 * relational_safety
         + 0.48 * bodily_regulation
         + 0.44 * memory_continuity
         + 0.40 * symbolic_recovery
         + 0.32 * witnessing_capacity
         - 0.68 * dissociation
         - 0.34 * nightmare_intrusion;
}

int main(void) {
    double score = integration_potential(0.82, 0.80, 0.84, 0.80, 0.78, 0.76, 0.78, 0.24, 0.28);
    printf("Synthetic trauma integration potential: %.3f\n", score);
    return 0;
}
